tests/test_visitors/test_ast/test_functions/test_getter_without_return.py

```diff
@@ -0,0 +1,249 @@
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    GetterWithoutReturnViolation,
+)
+from wemake_python_styleguide.visitors.ast.functions import (
+    FunctionSignatureVisitor,
+)
+
+getter_function_with_implicit_return = """
+def get_foo():
+    ...
+"""
+
+getter_function_with_bare_return = """
+def get_foo():
+    return
+"""
+
+getter_function_with_valued_return = """
+def get_foo():
+    return 1
+"""
+
+getter_function_with_explicit_none_return = """
+def get_foo():
+    return None
+"""
+
+getter_function_with_bare_yield = """
+def get_foo():
+    yield
+"""
+
+getter_function_with_valued_yield = """
+def get_foo():
+    yield 1
+"""
+
+getter_function_with_explicit_none_yield = """
+def get_foo():
+    yield None
+"""
+
+getter_function_with_yield_from = """
+def get_foo():
+    yield from [1]
+"""
+
+getter_method_with_implicit_return = """
+class Foo:
+    def get_foo():
+        ...
+"""
+
+getter_method_with_bare_return = """
+class Foo:
+    def get_foo():
+        return
+"""
+
+getter_method_with_valued_return = """
+class Foo:
+    def get_foo():
+        return 1
+"""
+
+getter_method_with_explicit_none_return = """
+class Foo:
+    def get_foo():
+        return None
+"""
+
+getter_method_with_bare_yield = """
+class Foo:
+    def get_foo():
+        yield
+"""
+
+getter_method_with_valued_yield = """
+class Foo:
+    def get_foo():
+        yield 1
+"""
+
+getter_method_with_explicit_none_yield = """
+class Foo:
+    def get_foo():
+        yield None
+"""
+
+getter_method_with_yield_from = """
+class Foo:
+    def get_foo():
+        yield from [1]
+"""
+
+regular_function_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_function_with_implicit_return = """
+class Foo:
+    def foo():
+        ...
+"""
+
+regular_function_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_function_with_bare_yield = """
+class Foo:
+    def foo():
+        yield
+"""
+
+regular_method_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_method_with_implicit_return = """
+class Foo:
+    def foo():
+        ...
+"""
+
+regular_method_with_bare_return = """
+class Foo:
+    def foo():
+        return
+"""
+
+regular_method_with_bare_yield = """
+class Foo:
+    def foo():
+        yield
+"""
+
+getter_method_with_branched_return = """
+def get_foo():
+    if bar:
+        return 1
+"""
+
+
+@pytest.mark.parametrize('code', [
+    getter_function_with_implicit_return,
+    getter_function_with_bare_return,
+    getter_function_with_bare_yield,
+    getter_method_with_implicit_return,
+    getter_method_with_bare_return,
+    getter_method_with_bare_yield,
```

Comment:
Bare `yield` should be allowed. Because `yield` can be used to define different constructs like `contextmanager` and `fixture`.

Can we please add this code as a test:

```python
def wrapper():
    print('context manager')
    yield
```

This should be allowed.

Reply:
that would be allowed because the function name doesn't start with `get_`. should I add the same function with name `get_wrapper`?

Reply:
Yeap 👍